src/game.zig

```diff
@@ -52,11 +52,41 @@ pub const camera = struct { // MARK: camera
 };
 
 pub const collision = struct {
-	pub fn triangleAABB(triangle: [3]Vec3d, box_center: Vec3d, box_extents: Vec3d) bool {
+	pub const AABB = struct {
+		min: Vec3d,
+		max: Vec3d,
+
+		pub fn center(self: AABB) Vec3d {
+			return (self.min + self.max)*@as(Vec3d, @splat(0.5));
+		}
+
+		pub fn extent(self: AABB) Vec3d {
+			return (self.max - self.min)*@as(Vec3d, @splat(0.5));
+		}
+
+		pub fn intersects(self: AABB, other: AABB) bool {
+			var i: usize = 0;
+			while(i < 3) : (i += 1) {
+				if(self.max[i] < other.min[i] or self.min[i] > other.max[i]) {
+					return false;
+				}
+			}
+			return true;
```

Comment:
Why not use vector ops?
```suggestion
			return @reduce(.And, (self.max >= other.min)) and @reduce(.And, (self.min <= other.max));
```